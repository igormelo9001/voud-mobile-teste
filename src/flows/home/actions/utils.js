import { pointsOfInterestTypes } from './enums';
import { calculateDistance } from '../../../utils/haversine';
import { getVouDRadiusFromViewableArea } from '../../../utils/get-radius-from-viewable-area';
import { HEADER_HEIGHT } from '../components/Header';
import { COLLAPSIBLE_HEIGHT } from '../components/ServicesMenu';
import { isArray } from 'util';

export const DEFAULT_LAT_DELTA = 0.025;
export const DEFAULT_LNG_DELTA = 0.025;

export function getTypesFromZoomLevel(longitudeDelta) {
	const zoomLevel = Math.round(Math.log(360 / longitudeDelta) / Math.LN2);

	if (zoomLevel >= 16) {
		return [
			pointsOfInterestTypes.STOP_TRAIN,
			pointsOfInterestTypes.STOP_BUS,
			pointsOfInterestTypes.STOP_SUBWAY,
			pointsOfInterestTypes.RECHARGE,
			pointsOfInterestTypes.BUS,
			pointsOfInterestTypes.BIKE,
			pointsOfInterestTypes.RENT_CAR
		];
	}
	if (zoomLevel < 16 && zoomLevel >= 14) {
		return [
			pointsOfInterestTypes.STOP_TRAIN,
			pointsOfInterestTypes.STOP_SUBWAY,
			pointsOfInterestTypes.RECHARGE,
			pointsOfInterestTypes.BUS,
			pointsOfInterestTypes.BIKE,
			pointsOfInterestTypes.RENT_CAR
		];
	}
	if (zoomLevel < 14 && zoomLevel >= 11) {
		return [
			pointsOfInterestTypes.STOP_TRAIN,
			pointsOfInterestTypes.STOP_SUBWAY,
			pointsOfInterestTypes.RECHARGE,
			pointsOfInterestTypes.BIKE,
			pointsOfInterestTypes.RENT_CAR
		];
	}
	if (zoomLevel < 11) {
		return [ pointsOfInterestTypes.RECHARGE, pointsOfInterestTypes.BIKE, pointsOfInterestTypes.RENT_CAR ];
	}

	return [];
}

const MAX_NUMBER_OF_VIEWABLE_POINTS = 200;
const MAX_NUMBER_OF_STORED_POINTS = 800;

export function mergeExistingPoints(currData, newData, maxNumber = MAX_NUMBER_OF_STORED_POINTS) {
	// Filter existing points on newPayload
	// if point id || externalId already exists,
	// remove it from final array
	const filteredNewData = newData.filter(
		(point) =>
			!currData.some((existingPoint) => {
				const pointId = point.id || point.externalId;
				const existingPointId = existingPoint.id || existingPoint.externalId;
				return pointId == existingPointId;
			})
	);

	const checkMap = {};
	const newArray = [ ...filteredNewData, ...currData ].filter((point) => {
		const checkedId = `${point.positionLatitude}_${point.positionLongitude}`;
		const checked = checkMap[checkedId];
		if (!checked) {
			checkMap[checkedId] = true;
		}
		return !checked;
	});

	return newArray.length > maxNumber ? newArray.splice(0, maxNumber) : newArray;
}

export function filterNearbyTembici(data, mapPosition) {
	if (!data) return [];

	const { latitude, longitude } = mapPosition;

	return data.sort(
		(a, b) =>
			calculateDistance(latitude, longitude, a.positionLatitude, a.positionLongitude) -
			calculateDistance(latitude, longitude, b.positionLatitude, b.positionLongitude)
	);
}

export function filterNearbyZazcar(data, mapPosition) {
	if (!data) return [];


	const { latitude, longitude } = mapPosition;

	data.sort(
	(a, b) =>
		calculateDistance(latitude, longitude, a.positionLatitude, a.positionLongitude) -
		calculateDistance(latitude, longitude, b.positionLatitude, b.positionLongitude)
	);

	const resultByAvailability = data.filter(
		(item) =>{
			if (item.pointType  !== pointsOfInterestTypes.RENT_CAR) return false;

			const carsCount = item.detail.filter(item => item.available == true).length;
			return carsCount > 0;
	});

	const result = data.filter(
		(item) =>{
			if (item.pointType  !== pointsOfInterestTypes.RENT_CAR) return false;

			return item.distance ==
			resultByAvailability.map((item) => Number.parseFloat(item.distance)).reduce((acc, cur) => acc && cur && Math.min(acc, cur))
		});

	if (isArray(result) && result.length > 0) {
		return result[0];
	}else{
		return null;
	}
}

export function filterPointsByZoomAndDistance(data, mapPosition, maxNumber = MAX_NUMBER_OF_VIEWABLE_POINTS) {
	if (!data) return [];

	const { latitude, longitude, latitudeDelta, longitudeDelta } = mapPosition;
	const regionRadius = getVouDRadiusFromViewableArea(latitude, longitude, latitudeDelta, longitudeDelta);
	const types = getTypesFromZoomLevel(longitudeDelta);

	const filteredArray = data
		.filter((point) => {
			const distance = calculateDistance(latitude, longitude, point.positionLatitude, point.positionLongitude);
			return distance < regionRadius && types.indexOf(point.pointType) >= 0;
		})
		.sort(
			(a, b) =>
				calculateDistance(latitude, longitude, a.positionLatitude, a.positionLongitude) -
				calculateDistance(latitude, longitude, b.positionLatitude, b.positionLongitude)
		);

	const selectedPoints = filteredArray.length > maxNumber ? filteredArray.splice(0, maxNumber) : filteredArray;

	return data.filter((point) =>
		selectedPoints.some((selectedPoint) => {
			const pointId = point.id || point.externalId;
			const selectedPointId = selectedPoint.id || selectedPoint.externalId;
			return pointId == selectedPointId;
		})
	);
}

export function filterCurrentDataByType(currData, types) {
	return currData.filter((point) => types.indexOf(point.pointType) >= 0);
}

export function shouldRenderDynamicPoints(longitudeDelta) {
	const types = getTypesFromZoomLevel(longitudeDelta);
	return types.indexOf(pointsOfInterestTypes.BUS) > 0;
}

export async function centerMarker(
	mapRef,
	mapHeight,
	menuCollapsed,
	latitude,
	longitude,
	latitudeDelta,
	longitudeDelta,
	duration = 400
) {
	if (!mapRef || !mapHeight || !latitude || !longitude || !latitudeDelta || !longitudeDelta) return;

	const bottomOffset = menuCollapsed ? 0 : latitudeDelta * (COLLAPSIBLE_HEIGHT / 2) / mapHeight;

	mapRef.animateToRegion(
		{
			latitude: latitude + latitudeDelta * (HEADER_HEIGHT / 2) / mapHeight - bottomOffset,
			longitude,
			latitudeDelta,
			longitudeDelta
		},
		duration
	);
}
