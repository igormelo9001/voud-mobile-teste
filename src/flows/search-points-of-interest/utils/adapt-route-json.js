export default function adaptRouteJson(item) {
    /*  It groups steps by type while maintaining the order
        and sums the total substeps of the route.*/
    const { steps } = item.legs[0];
    const stepsWithoutWalking = steps.filter((step) => {
      return step.travelMode !== "WALKING";
    });
    const formattedSteps = stepsWithoutWalking.reduce((acc, step, index) => {
      const { line } = step.transitDetails;
      const { type } = line.vehicle;
      const { shortName, name } = line;
      let stepObj = {};
      if(shortName) {
          stepObj.shortName = shortName.match(/\d+/)[0];
      } else {
        stepObj.shortName = name;
      }
      const { color } = line;
      if(color) {
        if(color === 'FFFF00') stepObj.color = '#F0C800';
        else stepObj.color = `#${color}`;
      }
      if (index === 0) {
        acc.steps.push({ type, substeps: [stepObj] });
        acc.totalSubsteps++;
        return acc;
      } else {
        const { length } = acc.steps;
        if (acc.steps[length - 1].type === type) {
          acc.steps[length - 1].substeps.push(stepObj);
        } else {
          acc.steps.push({ type, substeps: [stepObj] });
        }
        acc.totalSubsteps++;
        return acc;
      }
    }, { steps: [], totalSubsteps: 0 });
    return formattedSteps;
  }


  export function adaptColor(color){
    let newColor = ''; 
    if(color) {
      if(color === 'FFFF00') newColor = '#F0C800';
      else newColor =  color.indexOf('#') >= 0 ? color : `#${color}`;
    }
    return newColor;
  }