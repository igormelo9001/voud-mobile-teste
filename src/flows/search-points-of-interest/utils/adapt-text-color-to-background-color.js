import { colors } from "../../../styles";

export function adaptTextColorToBackgroundColor(backgroundColor) {
  if(backgroundColor === '#F0C800') return { color: colors.GRAY_DARKER };
  return { color: 'white' }
}