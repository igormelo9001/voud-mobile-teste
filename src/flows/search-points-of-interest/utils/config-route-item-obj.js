export default {
  TOTAL_UNAVAILABLE_WIDTH: 176,
  ABBREVIATED_SUBSTEP_ITEM_WIDTH: 60,
  SINGLE_SUBSTEP_ITEM_WIDTH: 82,      
  TWO_SUBSTEPS_ITEM_WIDTH: 127,
  INFINITE_SUBSTEPS_ITEM_WIDTH: 157,
}