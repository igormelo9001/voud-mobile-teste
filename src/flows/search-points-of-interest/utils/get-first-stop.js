export default function getFirstStop(item) {
  const { steps } = item.legs[0];
  return steps.find(step => step.travelMode !== 'WALKING');
}