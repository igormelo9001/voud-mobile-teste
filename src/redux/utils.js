/**
 * returns request status for any state
 * @return {Object} isFetching, requested, error, errorStatusCode
 */
export const extractRequestUI = obj => ({
  isFetching: obj.isFetching,
  requested: obj.requested || null,
  error: obj.error,
  errorStatusCode: obj.errorStatusCode || null,
  countAttempt: obj.countAttempt,
});

/**
 * returns merged request status for any state
 * @return {Object} isFetching, requested, error, errorStatusCode
 */
export const mergeRequestUI = objs =>
  objs
    .map(obj => extractRequestUI(obj))
    .reduce(
      (acc, cur) => ({
        isFetching: acc.isFetching ? true : cur.isFetching,
        requested: acc.requested ? acc.requested : cur.requested,
        error: cur.error ? cur.error : acc.error,
        errorStatusCode: cur.errorStatusCode ? cur.errorStatusCode : acc.errorStatusCode,
      }),
      {
        isFetching: false,
        requested: null,
        error: '',
        errorStatusCode: null,
      }
    );
