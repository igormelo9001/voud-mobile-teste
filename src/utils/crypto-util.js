import { encrypt } from "react-native-adyen-cse";
import { JSEncrypt } from "jsencrypt";

export const adyenEncrypt = async (
  { holder, cardNumber, securityCode, expirationDate },
  adyenPK
) => {
  const cardData = {
    holderName: holder,
    number: cardNumber,
    cvc: securityCode,
    expiryMonth: expirationDate.slice(0, 2),
    expiryYear: expirationDate.slice(3)
  };
  return await encrypt(cardData, adyenPK);
};

export const cieloEncrypt = (financialCardObj, cieloPK) => {
  const rsaEncrypt = new JSEncrypt();
  rsaEncrypt.setPublicKey(cieloPK);
  return rsaEncrypt.encrypt(JSON.stringify(financialCardObj));
};

export const buildEncryptedCardData = async (
  financialCardObj,
  cieloPK,
  adyenPK
) => {
  if (__DEV__) {
    console.tron.log(financialCardObj);
    console.tron.log("cieloPK");
    console.tron.log(cieloPK);

    console.tron.log("adyenPK");
    console.tron.log(adyenPK);
  }

  if (cieloPK && adyenPK) {
    return {
      encrypted: {
        CIELO: cieloEncrypt(financialCardObj, cieloPK),
        ADYEN: await adyenEncrypt(financialCardObj, adyenPK)
      }
    };
  } else {
    throw new Error("Ocorreu um erro, tente novamente em instantes.");
  }
};
