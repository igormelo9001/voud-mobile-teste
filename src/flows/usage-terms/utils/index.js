import {
  Platform
} from 'react-native';

import { openUrl } from '../../../utils/open-url';
import { navigateToRoute } from '../../../redux/nav';
import { routeNames } from '../../../shared/route-names';
import { fetchCheckLastTermsAccepted } from '../actions';

const showDocument = (dispatch, source) => {
  if (Platform.OS === 'ios') {
    dispatch(navigateToRoute(routeNames.BROWSER, { source }));
  } else {
    openUrl(source);
  }
};

export const viewTerms = dispatch => {
  const source = 'http://www.autopass.com.br/wp-content/uploads/2017/11/Termo-de-uso-app-VouD.pdf';
  showDocument(dispatch, source);
};

export const viewPrivacyPolicy = dispatch => {
  const source = 'http://www.autopass.com.br/wp-content/uploads/2017/11/Politica-de-privacidade-app-VouD.pdf';
  showDocument(dispatch, source);
};

export const dispatchCheckLastTermsAccepted = dispatch => {
  dispatch(fetchCheckLastTermsAccepted())
  .then(response => {
    if (response && !response.payload) dispatch(navigateToRoute(routeNames.ACCEPT_USAGE_TERMS));
  });
};
