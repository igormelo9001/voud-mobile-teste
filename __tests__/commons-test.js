// import React from 'react';
// import toJson from 'enzyme-to-json';

import { adyenEncrypt } from '../src/utils/crypto-util';

describe('Test Result from adyenEncrypt ', () => {
	it('should /src/components/adyenEncrypt result not undefined', async () => {
        const result = await adyenEncrypt(
			{ holder: 'Teste', cardNumber: 5555444433331111, securityCode: 737, expirationDate: '10/2020' },
            8415505105630389
        );
		expect(result).toMatchSnapshot();
    });
});
