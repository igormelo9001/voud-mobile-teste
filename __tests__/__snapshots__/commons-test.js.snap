// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Result from adyenEncrypt  should /src/components/adyenEncrypt result not undefined 1`] = `
Object {
  "cvc": 737,
  "expiryMonth": "10",
  "expiryYear": "2020",
  "holderName": "Teste",
  "number": 5555444433331111,
  "publicKey": 8415505105630389,
}
`;
